### ISSUE TITLE
Target [Illuminate\Contracts\Auth\Factory] is not instantiable.

### ISSUE THREAD

**Issue by SebastiaanKloos** [LukeTowers/oc-passport-plugin#3]
@LukeTowers Thank you for your awesome work on the Passport code. I get this error all the time. Do you know how we can fix this?

```
Target [Illuminate\Contracts\Auth\Factory] is not instantiable.
```

It is thrown everytime I use the 'auth' middleware. Do you have an example on how to use this on plugin routes?

**Comment by LukeTowers** [comment]
This plugin is still in beta, i.e. is not complete. I need to do some work integrating and testing this with https://github.com/octobercms/library/tree/laravel-auth and then we need to get those changes merged into October proper before we can have a solid implementation of Laravel's auth methods in October. Thanks for your interest it in however!

**Comment by SebastiaanKloos** [comment]
@LukeTowers Can't wait. I really like Passport! 😄

**Comment by numthang** [comment]
I have those problem with laravel-filemanager, too. Still in progress?

**Comment by LukeTowers** [comment]
@numthang no, work on it stopped since I don't currently have a need for it to be implemented and nobody else has a big enough need to either sponsor me to do the work or do it themselves.

**Comment by mohsin** [comment]
The reason it fails is that [Laravel Passport](https://github.com/laravel/passport/blob/4.0/src/RouteRegistrar.php#L48) requires the Authenticate middleware class in its route loader whereas [OctoberCMS disables that](https://github.com/octobercms/library/blob/master/src/Foundation/Http/Kernel.php#L39).

**Comment by b8ne** [comment]
FYI I seem to have solved this and got passport working correctly using the `auth:api` middleware/guard.
Implementation is as per this repo, however, I had to register the `Illuminate\Auth\Middleware\Authenticate` middleware somewhere.  
Did that but then that's where it throws this Factory is not instantiable error.
So I copied Authenticate.php into my own plugin and updated it to be as below:
```
<?php

use Closure;
use Illuminate\Auth\AuthenticationException;

class Authenticate
{
    /**
     * The authentication factory instance.
     */
    protected $auth;

    /**
     * Create a new middleware instance.
     *
     * @return void
     */
    public function __construct()
    {
        $this->auth = app()['auth'];
    }

    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Closure  $next
     * @param  string[]  ...$guards
     * @return mixed
     *
     * @throws \Illuminate\Auth\AuthenticationException
     */
    public function handle($request, Closure $next, ...$guards)
    {
        try {
            $this->authenticate($guards);
        } catch (AuthenticationException $e) {
            return response()->json(['error' => 'Unauthenticated!'], 401);
        }

        return $next($request);
    }

    /**
     * Determine if the user is logged in to any of the given guards.
     *
     * @param  array  $guards
     * @return void
     *
     * @throws \Illuminate\Auth\AuthenticationException
     */
    protected function authenticate(array $guards)
    {
        if (empty($guards)) {
            return $this->auth->authenticate();
        }

        foreach ($guards as $guard) {
            if ($this->auth->guard($guard)->check()) {
                return $this->auth->shouldUse($guard);
            }
        }

        throw new AuthenticationException('Unauthenticated.', $guards);
    }
}
```
Noting that `$this->auth` now references the auth instance in the IOC, not Factory.
Then added this to my plugin
```
public $middlewareAliases = [
        'auth' => Authenticate::class
    ];
protected function aliasMiddleware()
    {
        $router = $this->app['router'];

        $method = method_exists($router, 'aliasMiddleware') ? 'aliasMiddleware' : 'middleware';

        foreach ($this->middlewareAliases as $alias => $middleware) {
            $router->$method($alias, $middleware);
        }
    }
```
And all is working - including mocked tests using actingAs.

**Comment by LukeTowers** [comment]
@b8ne feel free to submit a pr with your changes if you'd like!
